src/tm_devices/drivers/pi/scopes/tekscope_2k/mso2kb.py

```diff
@@ -36,6 +43,102 @@ def __init__(
     ################################################################################################
     # Public Methods
     ################################################################################################
+    @property
+    def all_channel_names_list(self) -> Tuple[str, ...]:
+        """Return a tuple containing all the channel names.
+
+        Notes:
+            Includes the dedicated digital channel ``DCH1`` if ``MSO`` license is present.
+        """
+        return super().all_channel_names_list
+
+    @cached_property
+    def total_channels(self) -> int:
+        """Return the total number of channels (all types)."""
+        return 4
+
+    def curve_query(  # pylint: disable=too-many-locals
```

Comment:
Unless this is specific to the MSO2kB, please add this to the `TekScope2k` parent class so that all child classes can inherit it.

Reply:
Moved to TekScope2K